Add listening port customization and simple HTTP Basic authentication support

jqueuniet opened the pull request.

hispanico commented on templates/reverseproxy.conf.j2:
```diff
@@ -5,8 +5,8 @@
-    listen        80 default_server;
-    listen   [::]:80 default_server;
+    listen        {{ item.value.listen }} default_server;
+    listen   [::]:{{ item.value.listen }} default_server;
```
Missing listen default value

hispanico commented on tasks/main.yml:
```diff
@@ -10,6 +10,28 @@
+    state: present
+  tags:
+    - nginx-revproxy
+    - packages
```
Why do we need this packages?
As in the ansible docs, should't it need on the target host?

> The below requirements are needed on the host that executes this module.
> passlib>=1.6

hispanico commented on tasks/main.yml:
```diff
@@ -10,6 +10,28 @@
+    path: "/etc/nginx/{{ item.key }}_htpasswd"
+    name: "{{ item.value.auth.login }}"
+    password: "{{ item.value.auth.password }}"
+  loop: "{{ nginx_revproxy_sites|dict2items }}"
```
If we use loop with ```dict2items```, we need to change the Minimum Ansible Version to 2.6

hispanico commented:
Good job.
See my inline comments.

hispanico commented on tasks/main.yml:
```diff
@@ -10,6 +10,28 @@
+- name: Install python-passlib
+  apt:
+    name:
+        - "python-passlib"
```
Why do we need this packages?
As in the ansible docs, should't it need on the target host?

    The below requirements are needed on the host that executes this module.
    passlib>=1.6

jqueuniet commented:
Changed the defaults and the loop.

The passlib package is a requirement for the htpasswd module, has to be installed on any host targeted by this playbook, or else you get errors like this:

```failed: [worker-000] (item={'key': 'PUBLICIP_node-exporter', 'value': {'domains': ['PUBLICIP'], 'upstreams': [{'backend_address': '127.0.0.1', 'backend_port': 9001}], 'listen': 9002, 'auth': {'login': 'container', 'password': '****************'}, 'ssl': False, 'letsencrypt': False}}) => {"changed": false, "item": {"key": "{{ ansible_host }}_node-exporter", "value": {"auth": {"login": "container", "password": "***********"}, "domains": ["PUBLICIP"], "letsencrypt": false, "listen": 9002, "ssl": false, "upstreams": [{"backend_address": "127.0.0.1", "backend_port": 9001}]}}, "msg": "This module requires the passlib Python library"}```

I changed the install task to only pull the package for the relevant Python version.
